assets/src/dashboard/app/views/exploreTemplates/header/index.js

```diff
@@ -105,14 +104,40 @@ function Header({ filter, totalTemplates, search, templates, sort, view }) {
             onChange={onNewColorSelected}
           />
         </HeadingDropdownsContainer>
+      );
+    }
+    return null;
+  }, [
+    enableInProgressTemplateActions,
+    selectedCategories,
+    clearAllCategories,
+    onNewCategorySelected,
+    selectedColors,
+    clearAllColors,
+    onNewColorSelected,
+  ]);
+
+  return (
+    <Layout.Squishable>
+      <PageHeading
+        centerContent
+        defaultTitle={__('Templates', 'web-stories')}
+        searchPlaceholder={__('Search Templates', 'web-stories')}
+        stories={templates}
+        handleTypeaheadChange={
+          enableInProgressTemplateActions && search.setKeyword
+        }
+        typeaheadValue={search.keyword}
+      >
+        {TemplateFilters}
       </PageHeading>
       <BodyViewOptions
         resultsLabel={resultsLabel}
         layoutStyle={view.style}
         handleLayoutSelect={view.toggleStyle}
         currentSort={sort.value}
         pageSortOptions={TEMPLATES_GALLERY_SORT_MENU_ITEMS}
-        handleSortChange={sort.set}
+        handleSortChange={enableInProgressTemplateActions && sort.set}
```

Comment:
When `enableInProgressTemplateActions` is `false` the `&&` operator will short circuit and return `false` which is incompatible with the propType.

```
Warning: Failed prop type: Invalid prop `handleSortChange` of type `boolean` supplied to `BodyViewOptions`, expected `function`.
```

Removing the `isRequired` and setting this to `enableInProgressTemplateActions ? sort.set : undefined` would work.